components/server/src/user/user-service.ts

```diff
@@ -85,6 +85,16 @@ export class UserService {
         return result;
     }
 
+    async findTokensForIdentity(userId: string, identity: Identity): Promise<TokenEntry[]> {
+        const result = await this.userDb.findTokensForIdentity(identity);
+        for (const token of result) {
+            if (!(await this.authorizer.hasPermissionOnUser(userId, "read_info", token.uid))) {
```

Comment:
Should it return subset here?

Reply:
Identities belong to only one user. So if I can't see that user I can't find the tokens